R/measures.R

```diff
@@ -1415,27 +1415,25 @@ iauc.uno = makeMeasure(id = "iauc.uno", minimize = FALSE, best = 1, worst = 0,
 #' @rdname measures
 #' @format none
 ibrier = makeMeasure(id = "ibrier", minimize = TRUE, best = 0, worst = 1,
-  properties = c("surv", "req.truth", "req.model", "req.task"),
+  properties = c("surv", "req.prob", "req.feats"),
   name = "Integrated brier score using Kaplan-Meier estimator for weighting",
   note = "Only works for methods for which probabilities are provided via pec::predictSurvProb. Currently these are only coxph and randomForestSRC. To set an upper time limit, set argument max.time (defaults to max time in test data). Implemented in pec::pec",
   fun = function(task, model, pred, feats, extra.args) {
     requirePackages(c("survival", "pec"))
-    targets = getTaskTargets(task)
-    tn = getTaskTargetNames(task)
-    f = as.formula(sprintf("Surv(%s, %s) ~ 1", tn[1L], tn[2L]))
-    newdata = getTaskData(task)[model$subset, ]
-    max.time = extra.args$max.time %??% max(newdata[[tn[1L]]])
-    grid = seq(0, max.time, length.out = extra.args$resolution)
-
-    probs = predictSurvProb(getLearnerModel(model, more.unwrap = TRUE), newdata = newdata, times = grid)
-    perror = pec(probs, f, data = newdata[, tn], times = grid, exact = FALSE, exactness = 99L,
-      maxtime = max.time, verbose = FALSE, reference = FALSE)
-    
-
-    # FIXME: this might be the wrong number!
-    crps(perror, times = max.time)[1L, ]
-  },
-  extra.args = list(max.time = NULL, resolution = 1000L)
+    truth = getPredictionTruth(pred)
+    target = getPredictionTaskDesc(pred)$target
+    newdata = cbind(truth[,1], pred$data$truth.event, feats)
+    colnames(newdata)[1:2] = target
+    probs = getPredictionProbabilities(pred)
+    colnames(probs) = times.train = as.numeric(substr(colnames(probs), 11, 100))
```

Comment:
What does this do? It looks like this assumes the column names to be something particular.

Reply:
The training times are saved in the column names character. Maybe a bit weird, but I did not know where to save them elsewhere...

Reply:
Hmm, not sure if I understand. Where exactly are you saving training times, and what is the `substr()` doing?

Reply:
The training "survival times" are saved as column names of the probability predictions after "prob.time". So that's why "prob.time." is deleted first and then the rest is converted from character to numeric.

Reply:
If you have to save numeric values as strings, you are using the wrong data type(s). Such a conversion is not safe.

Reply:
Yes, but for example this works: paste0("x", 1:1000 + 10^-10)

It is a bit unlikely to have times smaller than 10^-13 ;). But see also my question above, where should I save the vector instead?